ingestion/src/util/tsupdate.rs

```diff
@@ -20,38 +26,97 @@ const OPEN_TIMESERIES_QUERY: &str = "\
     JOIN timeseries \
         ON met.timeseries = timeseries.id \
     WHERE met.param_id IS NOT NULL \
-    AND timeseries.totime IS NULL";
+    AND (timeseries.totime IS NULL \
+    OR timeseries.totime < timeseries.fromtime)";
+// NOTE: the from to in the timeseries table need to be kept updated
+// so we also need to check the from/to of the underlying data
+const MAX_MIN_TIMESERIES_DATA_QUERY: &str = "SELECT timeseries, 
+        MIN(obstime), \
+        MAX(obstime) \
+    FROM data \
+    WHERE timeseries = $1
+    GROUP BY timeseries";
+const MAX_MIN_TIMESERIES_NONSCALAR_DATA_QUERY: &str = "SELECT timeseries, 
+        MIN(obstime), \
+        MAX(obstime) \
+    FROM nonscalar_data \
+    WHERE timeseries = $1
+    GROUP BY timeseries";
 
+// Deactivated is information for the database
+// for a timeseries it is enough that the fromtime is closed
 const UPDATE_QUERY: &str = "\
     UPDATE public.timeseries SET \
-        totime = $1, \
-        deactivated = true \
-    WHERE id = $2";
+        fromtime = $1, \
+        totime = $2, \
+        deactivated = false \
+    WHERE id = $3";
 
-pub struct DeactivatedTimeseries {
+#[derive(Debug)]
+pub struct TSupdateTimeseries {
     /// Timeseries to be updated
     pub tsid: i64,
+    /// Fromtime value found in the metadata source
+    pub fromtime: DateTime<Utc>,
     /// Totime value found in the metadata source
     pub totime: DateTime<Utc>,
 }
```

Comment:
As suggested by my function suggestion above, I don't think we need this type, `(i64, OpenTimerange)` is fine IMO